### PR TITLE
Task_4: Created custom directives and pipes, added unit-tests and no-…

### PR DESCRIPTION
**Pull request by Russzh**
Created custom directives and pipes, added unit-tests and no-data block regarding [task_4](https://epam-my.sharepoint.com/personal/andrey_pavlov_epam_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fandrey%5Fpavlov%5Fepam%5Fcom%2FDocuments%2FAngularMentoring%2F4%5FDirectives%20and%20Pipes%2Epdf&parent=%2Fpersonal%2Fandrey%5Fpavlov%5Fepam%5Fcom%2FDocuments%2FAngularMentoring&ga=1)

### CODE REVIEW

File: my-app/angular.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
           "builder": "@angular-devkit/build-angular:browser",
           "options": {
             "allowedCommonJsDependencies": [
-              "lodash"
+              "moment"
```
Comment (aeroslav) [inline my-app/angular.json]: Please consider using date-fns or other alternatives. Moment is deprecated.

File: my-app/src/app/app.component.html [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
-<app-section></app-section>
-<app-courses
-  *ngFor="let iterCourse of courses; trackBy: trackByFn"
-  [course]="iterCourse"
-  (deleteCourse)="onDeleteCourse($event)"
->
-</app-courses>
-<app-load-more-button></app-load-more-button>
+<app-section (searchCourse)="onSearchCourse($event)"></app-section>
+<ng-container *ngIf="courses && courses.length !== 0">
```
Comment (aeroslav) [inline my-app/src/app/app.component.html:20]: Just `*ngIf="courses?.length"` should work

File: my-app/src/app/app.component.html [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 <app-breadcrumbs></app-breadcrumbs>
-<app-section></app-section>
-<app-courses
-  *ngFor="let iterCourse of courses; trackBy: trackByFn"
-  [course]="iterCourse"
-  (deleteCourse)="onDeleteCourse($event)"
->
-</app-courses>
-<app-load-more-button></app-load-more-button>
+<app-section (searchCourse)="onSearchCourse($event)"></app-section>
```
Comment (aeroslav) [inline my-app/src/app/app.component.html]: Looks strange. `app-section` seems to be something quite generic, kind of just a block or container. But suddenly it works with search functionality.

File: my-app/src/app/app.component.ts [hunk 4]
```diff
@@ -42,4 +29,12 @@ export class AppComponent implements OnInit {
   public trackByFn(index: number, item: Course): number {
     return item.id;
   }
+
+  public onSearchCourse(searchValue: string): void {
+    const searchValueTrimmed: string = searchValue.trim();
+    this.courses = this.filter.transform(this.courses, searchValueTrimmed);
```
Comment (aeroslav) [inline my-app/src/app/app.component.ts]: Let's discuss this approach on a meeting)

File: my-app/angular.json (hunk 1 above)
Comment (Russzh) [reply my-app/angular.json]: Moment was replaced with date-fns, thank you

File: my-app/src/app/app.component.ts (hunk 4 above)
Comment (Russzh) [reply my-app/src/app/app.component.ts]: Discussed

File: my-app/src/app/app.component.html (hunk 2 above)
Comment (Russzh) [reply my-app/src/app/app.component.html:20]: Discussed

File: my-app/src/app/app.component.html (hunk 3 above)
Comment (Russzh) [reply my-app/src/app/app.component.html]: Removed section component, added form and btn to app template, thank you

**Pull request merged**